components/ConditionalIf.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from 'react';
+import TruncateMarkup, { TruncateProps } from 'react-truncate-markup';
+import { Modal } from 'react-bootstrap';
+import { commonStyles } from '@htan/data-portal-commons';
+
+interface ConditionalIfValuesProps {
+    attribute: string;
+    attributes: string[];
+    truncateProps?: TruncateProps;
+}
+
+interface ViewAllValuesModalProps {
+    attribute: string;
+    options: JSX.Element[];
+    show: boolean;
+    onClose: () => void;
+}
+
+const ViewAllValuesModal: React.FunctionComponent<ViewAllValuesModalProps> = (
+    props
+) => {
+    return (
+        <Modal show={props.show} onHide={props.onClose}>
+            <Modal.Header closeButton>
+                <Modal.Title>{props.attribute} required if</Modal.Title>
+            </Modal.Header>
+            <Modal.Body>{props.options}</Modal.Body>
+        </Modal>
+    );
+};
+
+const ConditionalIfValues: React.FunctionComponent<ConditionalIfValuesProps> = (
+    props
+) => {
+    const [showModal, setShowModal] = useState(false);
+    const onModalClose = () => setShowModal(false);
+    const onClick = () => setShowModal(true);
+
+    const options = props.attributes
+        .filter((attribute) => attribute.length > 0)
+        .map((attribute) => (
+            <div key={attribute}>- {attribute.toLowerCase()}</div>
```

Comment:
I think it is okay to keep the original casing for Conditional If. We'll need to add an option like this
```suggestion
            <div key={attribute}>- {props.lowerCaseValues ? attribute.toLowerCase(): attribute}</div>
```